src/Illuminate/Database/Query/Builder.php

```diff
@@ -1749,21 +1747,24 @@ public function aggregate($function, $columns = ['*'])
         $this->bindings['select'] = $previousSelectBindings;
 
         if (isset($results[0])) {
-            return $this->formatAggregate($results);
+            return array_change_key_case((array) $results[0])['aggregate'];
         }
-
-        return 0;
     }
 
     /**
-     * Format the return value of an aggregate function.
+     * Execute and format an aggregate function on the database.
      *
-     * @param  array  $results
-     * @return float|int
+     * @param  string  $function
+     * @param  array   $columns
+     * @return mixed
      */
-    protected function formatAggregate($results)
+    public function formatedAggregate($function, $columns = ['*'])
```

Comment:
Probably make that `formattedAggregate` (typo fix) or `numberAggregate` for more descriptive name.